tm2/pkg/bft/privval/utils.go

```diff
@@ -25,14 +25,21 @@ func IsConnTimeout(err error) bool {
 }
 
 // NewSignerListener creates a new SignerListenerEndpoint using the corresponding listen address
-func NewSignerListener(listenAddr string, logger *slog.Logger) (*SignerListenerEndpoint, error) {
+func NewSignerListener(listenAddr string, logger *slog.Logger) (_ *SignerListenerEndpoint, rerr error) {
 	var listener net.Listener
 
 	protocol, address := osm.ProtocolAndAddress(listenAddr)
 	ln, err := net.Listen(protocol, address)
 	if err != nil {
 		return nil, err
 	}
+
+	defer func() {
+		if rerr != nil {
+			ln.Close()
+		}
```

Comment:
We shouldn't close this listener, otherwise the privval signer doesn't work

Reply:
if the returned error is non-nil, prival signer wouldn't be working. This issues named returns in Go

Reply:
`rerr` is never used aside from this function

Reply:
I get it now